src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -269,14 +268,44 @@ def __read_kv_from_app_service(cmd, appservice_account, prefix_to_add=""):
             cmd, resource_group_name=appservice_account["resource_group"], name=appservice_account["name"], slot=None)
         for item in settings:
             key = prefix_to_add + item['name']
-            value = item['value']
             tags = {'AppService:SlotSetting': str(item['slotSetting']).lower()} if item['slotSetting'] else {}
+            value = item['value']
+
+            # Value will look like one of the following if it is a KeyVault reference:
+            # @Microsoft.KeyVault(SecretUri=https://myvault.vault.azure.net/secrets/mysecret/ec96f02080254f109c51a1f14cdb1931)
+            # @Microsoft.KeyVault(VaultName=myvault;SecretName=mysecret;SecretVersion=ec96f02080254f109c51a1f14cdb1931)
```

Comment:
Is the schema fixed? It is possible to have whitespace, something like "@Microsoft.KeyVault(SecretUri = [https://myvault.vault.azure.net/secrets/mysecret/ec96f02080254f109c51a1f14cdb1931](https://myvault.vault.azure.net/secrets/mysecret/ec96f02080254f109c51a1f14cdb1931))"

Reply:
In appservice documentation, they haven't mentioned anything about whitespaces. To be on the safer side, I can strip all whitespaces before parsing. 
[https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references#reference-syntax](https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references#reference-syntax)  

---
In reply to: [364445905](https://github.com/Azure/azure-cli/pull/11773#discussion_r364445905) [](ancestors = 364445905)